modules/FIC_Github.py

```diff
@@ -114,3 +120,172 @@ def push(self):
+
+    def _get_date(self, commit):
+        self.commit_date = commit.commit.author.get("date")
+
+    def _get_author(self, commit):
+        self.commit_author = commit.commit.author.get("name")
+
+    def _get_author_email(self, commit):
+        self.commit_author_email = commit.commit.author.get("email")
+
+    def _get_url(self, commit):
+        self.commit_url = commit.url
+
+    def _get_files(self):
+        self.commit_files_changed = []
+        for item in (range(len(self.repo_data.commit(sha=self.commit_sha).files))):
+            self.commit_files_changed.append(self.repo_data.commit(sha=self.commit_sha).files[item].get('filename'))
+
+    def _compare_files(self):
+        for folder_to_check in range(len(self.folders_to_check)):
+            for changed_folder in range(len(self.commit_files_changed)):
+                if str(self.folders_to_check[folder_to_check]) in str(self.commit_files_changed[changed_folder]):
+                    return True
+
+    def _construct_commit(self):
+        self.list_of_commits.update({self.commit_number: {'sha': self.commit_sha,
+                                                          'url': self.commit_url,
+                                                          'author': self.commit_author,
+                                                          'author_email': self.commit_author_email,
+                                                          'message': self.commit_message,
+                                                          'date': self.commit_date,
+                                                          'files': self.commit_files_changed}})
+
+    def _commit_filter(self, commit_content):
+        if self.repo_type == "commit-keyword":
+            if self.keyword in self.commit_message:
+                self.commit_number += 1
+                self._store_data(commit_content)
+                self._construct_commit()
+
+        elif self.repo_type == "tag":
+            if self.repo_name == "build-puppet":
+                self.commit_number += 1
+                self._store_data(commit_content)
+                self._construct_commit()
+            elif self.release_version in self.commit_message:
+                self.commit_number += 1
+                self._store_data(commit_content)
+                self._construct_commit()
+
+        elif len(self.folders_to_check) > 0 and self._compare_files():
+            self.commit_number += 1
+            self._store_data(commit_content)
+            self._construct_commit()
+
+        else:
+            self.commit_number += 1
+            self._store_data(commit_content)
+            self._construct_commit()
+
+    def _not_tag(self):
+        self._last_checked()
+        self._commit_iterator()
+
+    def _build_puppet(self):
+        self._last_checked()
+        self._commit_iterator()
+
+    def _tag(self):
+        self._last_checked()
+        self._get_release()
+        if self.repo_name == "mozapkpublisher":
```

Comment:
This check can be done like this:
```Python
        if (self.repo_name == "mozapkpublisher") and (self.release_version != self.local_version):
            self._commit_iterator()
```

Reply:
It was done in: https://github.com/mozilla-releng/firefox-infra-changelog/pull/448/commits/16f80b00b7463d786334be5bc955e8dfac03b020